vfs/vfscache/downloaders/downloaders.go

```diff
@@ -386,6 +389,11 @@ func (dls *Downloaders) kickWaiters() (err error) {
 			fs.Errorf(dls.src, "vfs cache: restart download failed: %v", err)
 		}
 	}
+	if fserrors.IsErrNoSpace(dls.lastErr) {
+		fs.Errorf(dls.src, "vfs cache: cache is out of space %d/%d: last error: %v", dls.errorCount, maxErrorCount, dls.lastErr)
+		dls._closeWaiters(dls.lastErr)
```

Comment:
This returns the error to the waiters - I thought you wanted to avoid that?

Reply:
Don't we need the ENOSPC error to be passed back to the waiters so that the item._ensure() can get the error to start cache reset?

Reply:
That makes sense - ignore the above then :-)